kasa/device_factory.py

```diff
@@ -49,6 +49,20 @@ async def connect(*, host: Optional[str] = None, config: DeviceConfig) -> "Devic
     if host:
         config = DeviceConfig(host=host)
 
+    if (protocol := get_protocol(config=config)) is None:
+        raise UnsupportedDeviceException(
+            f"Unsupported device for {config.host}: "
+            + f"{config.connection_type.device_family.value}"
+        )
+
+    try:
+        return await _connect(config, protocol)
+    except:
+        await protocol.close()
+        raise
```

Comment:
```suggestion
    finally:
        await protocol.close()
```
This would also work and reads in my opinion nicer.

Reply:
That would cause the protocol to close when it's successfully been able to perform queries.  The desired behaviour is to close the protocol only if it's been unable to query the device and hence the `connect` call fails.

Reply:
Although, perhaps it's a good idea to add a test for this behavior pre-merge.

Reply:
Ok added assertions to the existing success and failure `connect` tests